exporter/uptraceexporter/exporter.go

```diff
@@ -58,9 +59,71 @@ func newTraceExporter(cfg *Config, logger *zap.Logger) (*traceExporter, error) {
 
 // pushTraceData is the method called when trace data is available.
 func (e *traceExporter) pushTraceData(ctx context.Context, traces pdata.Traces) (int, error) {
+	outSpans := make([]spanexp.Span, 0, traces.SpanCount())
+
+	rsSpans := traces.ResourceSpans()
+	for i := 0; i < rsSpans.Len(); i++ {
+		rsSpan := rsSpans.At(i)
+		resource := keyValueSlice(rsSpan.Resource().Attributes())
+
+		ils := rsSpan.InstrumentationLibrarySpans()
+		for j := 0; j < ils.Len(); j++ {
+			ilsSpan := ils.At(j)
+			lib := ilsSpan.InstrumentationLibrary()
+
+			spans := ilsSpan.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+
+				outSpans = append(outSpans, spanexp.Span{})
+				out := &outSpans[len(outSpans)-1]
+
+				out.ID = asUint64(span.SpanID().Bytes())
+				out.ParentID = asUint64(span.ParentSpanID().Bytes())
+				out.TraceID = span.TraceID().Bytes()
+
+				out.Name = span.Name()
+				out.Kind = spanKind(span.Kind())
+				out.StartTime = int64(span.StartTime())
+				out.EndTime = int64(span.EndTime())
+
+				out.Resource = resource
+				out.Attrs = keyValueSlice(span.Attributes())
+
+				out.StatusCode = statusCode(span.Status().Code())
+				out.StatusMessage = span.Status().Message()
+
+				out.TracerName = lib.Name()
+				out.TracerVersion = lib.Version()
+
+				out.Events = uptraceEvents(span.Events())
+				out.Links = uptraceLinks(span.Links())
+			}
+		}
+	}
+
+	if len(outSpans) == 0 {
+		return 0, nil
+	}
+
+	out := map[string]interface{}{
+		"spans": outSpans,
+	}
+	if err := e.upexp.SendSpans(ctx, out); err != nil {
+		if err, ok := err.(temporaryError); ok && err.Temporary() {
+			return len(outSpans), err
```

Comment:
Should this return a PartialTracesError?

Reply:
I believe it should not but I don't know much about it. Docs say:

```
// PartialError can be used to signalize that a subset of received data failed to be processed or send.
// The preceding components in the pipeline can use this information for partial retries.
```

But here all spans (not a subset) failed.